extensions/mapreduce/src/main/java/org/radargun/stages/mapreduce/MapReduceStage.java

```diff
@@ -121,20 +126,31 @@
    @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
    private MapReducer<KOut, VOut, R> mapReducer;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Clustered clustered;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private CacheInformation cacheInformation;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private BasicOperations basicOperations;
 
    @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Iterable iterable;
 
    private boolean supportsResultCacheName = false;
 
+   @Init
+   public void validate() {
```

Comment:
Ah, I see, because of the 'store result in cache' functionality. Hey, could you store the result temporarily in slave state and do the store operation in another stage? Ideally, I would remove the dependency between mapreduce and cache extensions alltogether.